### ISSUE TITLE
Trying to save scene cause viewport_set_canvas_stacking: Condition ' !viewport ' is true. 

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#23718]
**Godot version:**
3.1 66c6dfb

**OS/device including version:**
Windows 10

**Issue description:**
When I save scene from below then this error is shown:
```
ERROR: viewport_set_canvas_stacking: Condition ' !viewport ' is true.
   At: servers/visual/visual_server_viewport.cpp:541
```

**Steps to reproduce:**
1. Copy code from below and save as  Scene.tscn
2. Open this scene
3. Save this scene

**Minimal reproduction project:**
```
[gd_scene format=2]

[node name="Game" type="Node2D"]

[node name="Camera" type="Camera2D" parent="."]

[node name="HUD" type="CanvasLayer" parent="."]

```

**Comment by akien-mga** [comment]
CC @RandomShaper

**Comment by RandomShaper** [comment]
Oh, I think I now what's happening. I'll fix it (as well as the 3.0 PR) in a few hours.
